### DISCUSSION TITLE
Issue with samsung tv

### DISCUSSION THREAD

**Discussion by freerideguy** [vicwomg/pikaraoke#325] [Q&A]
this is killer for our home fun-room. Wish i saw this back in the Covid years.  I do have one issue that I can't seem to figure out.  On a samsung tv there is only one browser option.  When I use the headless option and enter the IP from the rasp it loads the main screen stating that I need to interact with the website first and click ok but on the samsung tv it never lets me click on anything.  i've tried the remote/keyboard/mouse.  on every other browser that i have tested it works fine.   of course the samsung tv is the one that i want to use and can't.  Is there an option to somehow bypass that screen and go right to the main splash screen?  
![karaoke issue](https://github.com/vicwomg/pikaraoke/assets/131824726/419c7298-d75d-442a-afbb-3ef62b03efe0)

**Comment by vicwomg** [comment]
Oh, I 'm actually very curious about these embedded browsers! So there's no way to click buttons within them? Have you tried on other websites? 

I just pushed a change to master that adds the `/splash.html?confirm=false` parameter to the splash screen url which will skip this step. Pull the latest code to try it. Maybe the browser is not standards-compliant and will let the video autoplay.  Would be curious to know.

**Reply by vicwomg** [reply]
Commit: https://github.com/vicwomg/pikaraoke/commit/e546b2839e05ae21c64a8adfaf2b1686b907d8a0

**Reply by freerideguy** [reply]
I have a Q version of samsung tv.  no option to enable link browsing on the tv that I can see.  it's also not a model that can be "rooted" by the sammy software.  The samsung defualt browser is very useless, slow and clunky.  But to directly answer the question when on normal web pages like google search or whatnot, when the curser scans over a clickable link, it will allow me to click the remote, then load the page.  

let me start by warning you I am not good at this stuff -it took me 3 days to figure out how to install this - im betting the normal computer guys can do it in 5min.  Hopefully my answers make sense?  But i may not be speaking the right lingo. 

It behaved normally when booted up to the PIkaraoke, loads the splash screen, and highlights the "confirm" button but when clicking on the remote it does not do anything.  Same with a USB mouse and keyboard. 

Just ran update - running 1.2 and still same behavior .  will not let me past the confirm screen on the tv, but no issues on phone/tablet/computer.   I noticed there is a dropdown box in the upper left corner that is also not responsive and won't let me click on it.  

I am going to test it on a Nvidia Shield, with same tv.  I am assuming it will work but just want to know. 

thanks for the fast reply - I always wonder if these posts are dead.  glad to see this one is alive and kicking!

**Reply by vicwomg** [reply]
@freerideguy Ok, got it. I'll try to give more specific instructions

When you say you updated it, how did you update the code? To be clear, to update pikaraoke to the latest, you go to the pikaraoke directory in the terminal, then type `git pull`, which will pull down the latest code. Then you need to restart pikaraoke. 

Also, the change I added allows you to tack on `confirm=false` to the end of the splash screen URL to skip the confirmation modal. So it will still show unless you do that. Example, you would enter this URL instead to the TV: `https://<ip_of_your_pikaraoke>:5555/splash?confirm=false`. If you do that it will hide the confirmation prompt, but there's no guarantee it will work on the TV because it basically skips this necessary step. On most browsers you must interact with the page once to allow autoplaying videos. However it's possible the Samsung browser doesn't have this restriction.

Your comments also gave a good hint: links do work on the browser. So I also just pushed an update that adds a link to the confirmation prompt which might be better recognized by the embedded browser (so you can click the "confirm" text as well as the button. Give that a try too (again, to pull the latest, run `git pull` from the pikaraoke directory)

<img width="811" alt="Screenshot 2024-03-30 at 3 38 30 PM" src="https://github.com/vicwomg/pikaraoke/assets/4107190/dfa8f979-de7c-420a-baa0-4bb375d3e256">

**Comment by vicwomg** [comment]
Also, I'm wondering if you need to enable "link browsing" on the TV's browser to actually select stuff:

https://www.samsung.com/in/support/tv-audio-video/how-to-turn-on-link-browsing-in-samsung-f-series-smart-tv/#:~:text=1%20Connect%20TV%20to%20internet%20and%20check%20the%20Network%20Status.&text=2%20After%20establishing%20the%20internet,better%20experience%20on%20Web%20browser.
